research/src/pages/component-spec-template.mdx

```diff
@@ -0,0 +1,153 @@
+> | --------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+> | Up/Right  | Increments the value of the {ComponentName} by the amount specified by the `step` prop. If the `shift` modifier is pressed, increases by 10x the `step` value, clamped within `min`/`max` values. |
+> | Down/Left | Decrements the value of the {ComponentName} by the amount specified by the `step` prop. If the `shift` modifier is pressed, decreases by 10x the `step` value, clamped within `min`/`max` values. |
+
+### Mouse interaction
+
+> Define the mouse behaviors of the component.
+>
+> - What specific mouse events (down, up, move) are being listened to and how does that affect the component?
+> - For behaviors which are accessible through mouse input, can you also do the same thing using a keyboard?
+>
+> Example:
+>
+> - `mousedown` should immediately attempt to set value to the appropriate value.
+> - `mousemove` should be attached to the window during mousedown to track window-wise move events. Value should be updated appropriately. Note; if a `mousemove` occurs
+>   without the primary button pressed, tracking should cancel and treat the even as a `mouseup`. (Edge case: mouse down, move cursor out of window, release mouse)
+> - `mouseup` should remove the `mousemove` event.
+
+### Touch interaction
+
+> Define the touch behaviors of the component.
+>
+> - Are there different behaviors than in mouse scenarios?
+> - Are touch targets large enough for fat fingers? (44x44px is the recommended minimum size.) For example, the `thumb` of a slider might appear to be small, but the touchable area should be larger.
+
+### Screen reader accessibility
+
+> Review/link aria specifications and list which aria labels should appear on which parts of the component.
+>
+> If there are ambiguous requirements or lacking guidance in the specifications, consult the experts in your organization to propose refinements/additions to the specifications.
+>
+> Example:
+>
+> `root`:
+>
+> - should render the native element using the `as` prop, defaulting to `div`
+> - should mix in native props expected for the element type defined in `as`.
+>
+> `thumb`:
+>
+> - should be focusable via `tabindex=0`
+> - role set to `{ComponentName}`
+> - receives `aria-valuemin` and `aria-valuemax` representing min/max
+> - receives `aria-valuenow` representing the current value
+
+### Accessibility concerns for the user
+
+> Consider which props, mapped to attribute values, must be provided for the user to make the component accessible. Often props which require localization or calculating a value must be provided by the user.
+>
+> For React props, consider using `ariaLabel` and `ariaLabelledBy`, rather than the natural kebab-cased variants (`aria-label`) so that they can be provided without destroying type safety.
```

Comment:
@dzearing I'm torn on this. In our implementation we extend existing type interfaces for DOM elements and still get type safety. Is the important thing here to ensure that these can be enumerated, or that the implementation is 1:1? I would think that the goal is less the details of the interface (how we accomplish passing an aria-label) and more as to whether it is *required* or *beneficial* for a given implementation.

Reply:
Also, should we be getting into any react specifics? For example, shouldn't we just use the standardized variant if it is a standardized attribute and the implementor can map this to their framework/lib or web component.

Reply:
Yeah, I think we should try as hard as possible to stay implementation agnostic - aligning to web standards will help with that and allow us to scale as the next new, cool thing comes out.

Reply:
good calls